docs/how-to-guides/data-basics/5-adding-a-delete-button.md

```diff
@@ -0,0 +1,454 @@
+
+Here's how you can try it in your browser's dev tools:
+
+```js
+// We need a valid page ID to call deleteEntityRecord, so let's get the first available one using getEntityRecords.
+const pageId = wp.data.select( 'core' ).getEntityRecords( 'postType', 'page' )[0].id;
+
+// Now let's delete that page:
+const promise = wp.data.dispatch( 'core' ).deleteEntityRecord( 'postType', 'page', pageId );
+
+// promise gets resolved or rejected when the API request succeeds or fails.
+```
+
+Once the REST API request is finished, you will notice one of the pages has disappeared from the list. This is because that list is populated by using `useSelect()` and the `select( coreDataStore ).getEntityRecords( 'postType', 'page' )` selector. Anytime the underlying data changes, the list gets re-rendered with fresh data. That's pretty convenient!
+
+Let's dispatch that action when `DeletePageButton` is clicked:
+
+```js
+const DeletePageButton = ({ pageId }) => {
+	const { deleteEntityRecord } = useDispatch( coreDataStore );
+	const handleDelete = () => deleteEntityRecord( 'postType', 'page', pageId );
+	return (
+		<Button variant="primary" onClick={ handleDelete }>
+			Delete
+		</Button>
+	);
+}
+```
+
+### Step 3: Add visual feedback
+
+It may take a few moments for the REST API request to finish after clicking the _Delete_ button. Let's communicate that with a `<Spinner />` component similarly to what we did in the previous parts of this tutorial.
+
+We'll need the `isDeletingEntityRecord` selector for that. It is similar to the `isSavingEntityRecord` selector we've already seen in [part 3](/docs/how-to-guides/data-basics/3-building-an-edit-form.md): it returns `true` or `false` and never issues any HTTP requests:
+
+```js
+const DeletePageButton = ({ pageId }) => {
+	// ...
+	const { isDeleting } = useSelect(
+		select => ({
+			isDeleting: select( coreDataStore ).isDeletingEntityRecord( 'postType', 'page', pageId ),
+		}),
+		[ pageId ]
+	)
+	return (
+		<Button variant="primary" onClick={ handleDelete } disabled={ isDeleting }>
+			{ isDeleting ? (
+				<>
+					<Spinner />
+					Deleting...
+				</>
+			) : 'Delete' }
+		</Button>
+	);
+}
+```
+
+Here's what it looks like in action:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/deleting-in-progress.png)
+
+### Step 4: Handle errors
+
+We optimistically assumed that a *delete* operation would always succeed. Unfortunately, under the hood, it is a REST API request that can fail in many ways:
+
+* The website can be down.
+* The update may be invalid.
+* The page could have been deleted by someone else in the meantime.
+
+To tell the user when any of these happens, we need to extract the error information using the `getLastEntityDeleteError` selector:
+
+```js
+// Replace 9 with an actual page ID
```

Comment:
This is a bit confusing. Why not use a placeholder instead of this comment about the magic number?

Reply:
How would you introduce a placeholder that would clearly communicate that?

Reply:
I left the placeholder for now, I'm happy to revisit later on, though!